engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -1046,17 +1046,14 @@ static int CreateTextureAsync(lua_State* L)
     CreateTextureResourceParams create_params = {};
     CheckCreateTextureResourceParams(L, &create_params);
 
-    // Create an empty upload buffer
-    dmBuffer::HBuffer upload_buffer = create_params.m_Buffer;
-    bool use_upload_buffer = create_params.m_Buffer == 0;
-    if (use_upload_buffer)
+    // We need to create an empty upload buffer if no explicit buffer is passed
+    bool is_transcoded              = create_params.m_CompressionType != dmGraphics::TextureImage::COMPRESSION_TYPE_DEFAULT && dmGraphics::IsFormatTranscoded(create_params.m_CompressionType);
+    bool use_raw_data               = create_params.m_Buffer == 0 || is_transcoded;
+    uint8_t* raw_data               = 0;
+
+    if (create_params.m_Buffer == 0)
     {
-        const dmBuffer::StreamDeclaration streams_decl = {dmHashString64("data"), dmBuffer::VALUE_TYPE_UINT8, 1};
-        dmBuffer::Result r = dmBuffer::Create(create_params.m_Width * create_params.m_Height * create_params.m_TextureBpp, &streams_decl, 1, &upload_buffer);
-        if (r != dmBuffer::RESULT_OK)
-        {
-            return DM_LUA_ERROR("Unable to create an empty upload buffer: %s (%d)", dmBuffer::GetResultString(r), r);
-        }
```

Comment:
Not sure why we used an upload buffer and not just a raw data pointer 🤔 
I guess in some sense we could use the same code path, but it _feels_ like it was just doing a malloc with extra steps. We don't really need the alignment, streams etc.

Reply:
Isn't it about ownership? Lua vs Resource system or something like this. I don't remember for sure, I just remember we had conversation around it

Reply:
But this buffer will never be placed into lua, it's just an intermediate storage for the texture data during async upload.. The other buffer that can come from lua is handled according to this, we increase the ref while the data is uploaded and then its decref'd when upload is done. But in this case I Don't really see why we use an intermediate HBuffer 🤔